@VindeeR :

```
=========================== short test summary info ============================
FAILED test/integration/svn/test_autosubmit_svn.py::test_svn_submodules_dirty - autosubmit.log.log.AutosubmitCritical: Can not check out revision 1 http://localhost:39121/svn/svn-project into /tmp/pytest-of-runner/pytest-0/popen-gw0/test_svn_submodules_dirty0/t01m/proj
```

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -104,7 +104,7 @@ jobs:
           cache: 'pip' # caching pip dependencies
 
       - name: Install system dependencies
-        run: sudo apt-get install -y curl git graphviz rsync xvfb r-base
+        run: sudo apt-get install -y curl git subversion graphviz rsync xvfb r-base
 
       - name: Install dependencies
         run: |

--- a/autosubmit/autosubmit.py
+++ b/autosubmit/autosubmit.py
@@ -4430,9 +4430,15 @@ class Autosubmit:
             Log.debug(f"The project folder {local_proj_dir} has been created.")
             Log.info(f"Checking out revision {svn_project_revision + ' ' + svn_project_url} into {local_proj_dir}")
             try:
-                output = subprocess.check_output("cd " + local_proj_dir + "; svn --force-interactive checkout -r " +
-                                                 svn_project_revision + " " + svn_project_url + " " +
-                                                 project_destination, shell=True)
+                cmd = (f"cd {local_proj_dir}; svn --force-interactive checkout -r {svn_project_revision} "
+                       f"{svn_project_url} {project_destination}")
+                svn_user = as_conf.experiment_data.get("CUSTOM_CONFIG", "").get("USER")
+                if svn_user is not None:
+                    cmd += f" --username {svn_user}"
+                svn_password = as_conf.experiment_data.get("CUSTOM_CONFIG", "").get("PASSWORD")
+                if svn_password is not None:
+                    cmd += f" --password {svn_password}"
+                output = subprocess.check_output(cmd, shell=True)
             except subprocess.CalledProcessError:
                 try:
                     shutil.rmtree(local_proj_dir, ignore_errors=True)

--- a/pytest.ini
+++ b/pytest.ini
@@ -17,6 +17,7 @@ markers =
     docker: marks a test that requires docker to run.
     slurm: marks a test that requires a Slurm container to run.
     git: marks a test that requires a Git server to run.
+    svn: marks a test that requires a SVN server to run.
     postgres: mark a test that requires postgres to run.
     ssh: marks a test that requires an SSH server.
     x11: marks a test that requires an SSH server with X11 enabled.

--- a/test/integration/conftest.py
+++ b/test/integration/conftest.py
@@ -326,7 +326,7 @@ def paramiko_platform() -> Iterator[ParamikoPlatform]:
 
 @pytest.fixture(scope="function")
 def git_server(request, tmp_path) -> Generator[tuple['DockerContainer', Path, str], Any, None]:
-    # Start a container to server it -- otherwise, we would have to use
+    # Start a container to serve it -- otherwise, we would have to use
     # `git -c protocol.file.allow=always submodule ...`, and we cannot
     # change how Autosubmit uses it in `autosubmit create` (due to bad
     # code design choices).
@@ -347,7 +347,7 @@ def git_server(request, tmp_path) -> Generator[tuple['DockerContainer', Path, st
 
 @pytest.fixture(scope="function")
 def svn_server(request, tmp_path) -> Generator[tuple['DockerContainer', Path, str], Any, None]:
-    # Start a container to server it -- otherwise, we would have to use
+    # Start a container to serve it -- otherwise, we would have to use
     # `svn -c protocol.file.allow=always submodule ...`, and we cannot
     # change how Autosubmit uses it in `autosubmit create` (due to bad
     # code design choices).

--- a/test/integration/conftest.py
+++ b/test/integration/conftest.py
@@ -47,7 +47,7 @@ from autosubmit.platforms.slurmplatform import SlurmPlatform
 from test.integration.test_utils.docker import (
     get_git_container, get_slurm_container, get_ssh_container,
     prepare_and_test_slurm_container, prepare_and_test_ssh_container,
-    prepare_and_test_git_container
+    prepare_and_test_git_container, get_svn_container, prepare_and_test_svn_container
 )
 from test.integration.test_utils.networking import get_free_port
 from test.integration.test_utils.postgres import setup_pg_db
@@ -343,6 +343,27 @@ def git_server(request, tmp_path) -> Generator[tuple['DockerContainer', Path, st
         prepare_and_test_git_container(container, http_port)
         yield container, git_repos_path, repo_url
         copy_content_from_containers(request, 'git_server', '/opt/git-server')
+
+
+@pytest.fixture(scope="function")
+def svn_server(request, tmp_path) -> Generator[tuple['DockerContainer', Path, str], Any, None]:
+    # Start a container to server it -- otherwise, we would have to use
+    # `svn -c protocol.file.allow=always submodule ...`, and we cannot
+    # change how Autosubmit uses it in `autosubmit create` (due to bad
+    # code design choices).
+    base_path = tmp_path / 'svn_repos_base'
+    svn_repos_path = base_path / 'svn_repos'
+    svn_repos_path.mkdir(exist_ok=True, parents=True)
+
+    container, http_port = get_svn_container(svn_repos_path)
+    # http_port = int(container.ports['80/tcp'][0]['HostPort'])  # type: ignore
+
+    repo_url = f'http://localhost:{http_port}/svn'
+
+    with container:
+        prepare_and_test_svn_container(container, http_port)
+        yield container, svn_repos_path, repo_url
+        # copy_content_from_containers(request, 'svn_server', 'svn-project')
 
 
 @pytest.fixture

--- a/test/integration/svn/__init__.py
+++ b/test/integration/svn/__init__.py
@@ -1,0 +1,18 @@
+# Copyright 2015-2025 Earth Sciences Department, BSC-CNS
+#
+# This file is part of Autosubmit.
+#
+# Autosubmit is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Autosubmit is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Autosubmit.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Integration tests for Autosubmit SVN functions."""

--- a/test/integration/svn/test_autosubmit_svn.py
+++ b/test/integration/svn/test_autosubmit_svn.py
@@ -68,21 +68,18 @@ def test_svn_submodules_dirty(
 
     svn_repo = svn_repos_path / 'svn-project'
 
-    svn_repo_url = f'{svn_url}/{svn_repo.name}'
-
     experiment_data = _get_experiment_data()
     experiment_data['PROJECT']['PROJECT_TYPE'] = 'svn'
-    experiment_data['PROJECT']['PROJECT_DESTINATION'] = svn_repo.name
-    experiment_data['SVN']['PROJECT_URL'] = svn_repo_url
+    experiment_data['SVN']['PROJECT_URL'] = f'{svn_url}/{svn_repo.name}'
     experiment_data['SVN']['PROJECT_REVISION'] = '1'
     experiment_data['CUSTOM_CONFIG']['USER'] = 'svnadmin'
     experiment_data['CUSTOM_CONFIG']['PASSWORD'] = 'test'
 
-    as_exp = autosubmit_exp(experiment_data=experiment_data)
-    as_conf = as_exp.as_conf
+    as_exp = autosubmit_exp('t001', experiment_data=experiment_data)
+    proj_dir = Path(as_exp.as_conf.get_project_dir())
 
-    proj_dir = Path(as_conf.get_project_dir())
-
-    assert (proj_dir / 'LOCAL_SETUP.sh').exists()
-    assert (proj_dir / 'REMOTE_SETUP.sh').exists()
+    assert proj_dir.parts[-1] == 'svn-project'
+    assert (proj_dir / 'branches').exists()
+    assert (proj_dir / 'tags').exists()
+    assert (proj_dir / 'trunk').exists()
 

--- a/test/integration/svn/test_autosubmit_svn.py
+++ b/test/integration/svn/test_autosubmit_svn.py
@@ -1,0 +1,88 @@
+# Copyright 2015-2025 Earth Sciences Department, BSC-CNS
+#
+# This file is part of Autosubmit.
+#
+# Autosubmit is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Autosubmit is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Autosubmit.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Integration tests for ``autosubmit_svn``."""
+from pathlib import Path
+from typing import Callable, TYPE_CHECKING
+
+import pytest
+
+if TYPE_CHECKING:
+    from docker.models.containers import Container
+
+
+# svn operational check
+
+def _get_experiment_data() -> dict:
+    return {
+        'JOBS': {
+            'debug': {
+                'SCRIPT': 'echo "Hello world"',
+                'RUNNING': 'once'
+            },
+        },
+        'PROJECT': {
+            'PROJECT_TYPE': 'svn',
+            'PROJECT_DESTINATION': 'svn-project',
+        },
+        'SVN': {
+        },
+        'CUSTOM_CONFIG':{
+            'USER': 'svnadmin',
+            'PASSWORD': 'test',
+        },
+    }
+
+
+@pytest.mark.svn
+@pytest.mark.docker
+def test_svn_submodules_dirty(
+        autosubmit_exp: Callable,
+        svn_server: tuple['Container', Path, str],
+        tmp_path
+) -> None:
+    """Tests that Autosubmit detects dirty local svn submodules, especially with operational experiments.
+
+    This test has a svn repository with a svn submodule. The parameters in this test control whether the
+    svn repository and the svn submodule contents will be committed and pushed.
+
+    If the user has non-committed or non-pushed changes in the repository or submodule, the code is
+    expected to fail, raising an error when the experiment is operational.
+    """
+
+    container , svn_repos_path, svn_url = svn_server  # type: Container, Path, str # type: ignore
+
+    svn_repo = svn_repos_path / 'svn-project'
+
+    svn_repo_url = f'{svn_url}/{svn_repo.name}'
+
+    experiment_data = _get_experiment_data()
+    experiment_data['PROJECT']['PROJECT_TYPE'] = 'svn'
+    experiment_data['PROJECT']['PROJECT_DESTINATION'] = svn_repo.name
+    experiment_data['SVN']['PROJECT_URL'] = svn_repo_url
+    experiment_data['SVN']['PROJECT_REVISION'] = '1'
+    experiment_data['CUSTOM_CONFIG']['USER'] = 'svnadmin'
+    experiment_data['CUSTOM_CONFIG']['PASSWORD'] = 'test'
+
+    as_exp = autosubmit_exp(experiment_data=experiment_data)
+    as_conf = as_exp.as_conf
+
+    proj_dir = Path(as_conf.get_project_dir())
+
+    assert (proj_dir / 'LOCAL_SETUP.sh').exists()
+    assert (proj_dir / 'REMOTE_SETUP.sh').exists()
+

--- a/test/integration/test_utils/docker.py
+++ b/test/integration/test_utils/docker.py
@@ -45,7 +45,9 @@ __all__ = [
     'prepare_and_test_slurm_container',
     'get_ssh_container',
     'prepare_and_test_ssh_container',
-    'stop_test_containers'
+    'stop_test_containers',
+    'get_svn_container',
+    'prepare_and_test_svn_container'
 ]
 
 _SSH_DOCKER_IMAGE = 'lscr.io/linuxserver/openssh-server:latest'
@@ -185,9 +187,14 @@ def prepare_and_test_svn_container(container: 'DockerContainer', http_port: int)
     container.exec("sed -i 's/= r/= rw/g' /etc/subversion/subversion-access-control")
     container.exec("svnadmin create home/svn/svn-project")
     container.exec("chown -R apache:apache /home/svn/svn-project")
-    container.exec("touch /home/svn/svn-project/LOCAL_SETUP.sh")
-    container.exec("touch /home/svn/svn-project/REMOTE_SETUP.sh")
-    container.exec("svn mkdir file:///home/svn/svn-project/trunk file:///home/svn/svn-project/branches file:///home/svn/svn-project/tags file:///home/svn/svn-project/LOCAL_SETUP.sh file:///home/svn/svn-project/REMOTE_SETUP.sh -m 'first commit'")
+    container.exec(
+        "svn mkdir "
+        "file:///home/svn/svn-project/trunk "
+        "file:///home/svn/svn-project/branches "
+        "file:///home/svn/svn-project/tags "
+        "-m 'init dirs'"
+    )
+    container.exec("echo -e 'store-plaintext-passwords = yes' >> /home/.subversion/servers")
 
     wait_for_tcp_port('localhost', http_port)
 

--- a/test/integration/test_utils/docker.py
+++ b/test/integration/test_utils/docker.py
@@ -60,11 +60,15 @@ _SLURM_DOCKER_IMAGE = 'giovtorres/slurm-docker:25.11.2-v0.1.7'
 
 _GIT_DOCKER_IMAGE = 'githttpd/githttpd:latest'
 """The Git image used for tests where Autosubmit needs to clone a repository."""
+_SVN_DOCKER_IMAGE = 'elleflorio/svn-server:latest'
+"""The SVN image used for tests where Autosubmit needs to clone a repository."""
 
 _AS_SLURM_CONTAINER_LABEL = "pytest.slurm.singleton"
 """Docker container label key for Slurm singleton."""
 _AS_GIT_CONTAINER_LABEL = "pytest.git.singleton"
 """Docker container label key for Git singleton."""
+_AS_SVN_CONTAINER_LABEL = "pytest.svn.singleton"
+"""Docker container label key for SVN singleton."""
 _AS_SSH_CONTAINER_LABEL = "pytest.ssh.singleton"
 """Docker container label key for SSH singleton."""
 _AS_SSH_X11_CONTAINER_LABEL = "pytest.ssh_x11.singleton"
@@ -145,6 +149,54 @@ def get_git_container(git_repos_path: Path) -> tuple['DockerContainer', int]:
     http_port = get_free_port()
     # noinspection PyProtectedMember
     container_instance = _create_git_container(git_repos_path, http_port)
+    return container_instance, http_port
+
+
+def _create_svn_container(svn_repos_path: Path, http_port: int) -> DockerContainer:
+    """Start a Docker container with Git.
+
+    The repository will be available with the base name of the path given,
+    with the TCP/80 port mapped to the host ``http_port``.
+    """
+    docker_args = {
+        'labels': {
+            _AS_SVN_CONTAINER_LABEL: 'true',
+        }
+    }
+
+    docker_container = DockerContainer(
+        image=_SVN_DOCKER_IMAGE,
+        remove=True,
+        **docker_args
+    )
+
+    container = docker_container \
+        .with_bind_ports(80, http_port) \
+        .with_volume_mapping(str(svn_repos_path), '/home/svn', mode='rw')
+
+    # The docker image ``elleflorio/svn-server`` creates an HTTP server for SVN
+    # repositories, using the volume bound onto ``<TBC>`` as base
+    # for any subdirectory, the SVN URL becoming ``svn/{subdirectory-name}}``.
+    return container
+
+
+def prepare_and_test_svn_container(container: 'DockerContainer', http_port: int) -> None:
+    container.exec("htpasswd -b /etc/subversion/passwd svnadmin test")
+    container.exec("sed -i 's/= r/= rw/g' /etc/subversion/subversion-access-control")
+    container.exec("svnadmin create home/svn/svn-project")
+    container.exec("chown -R apache:apache /home/svn/svn-project")
+    container.exec("touch /home/svn/svn-project/LOCAL_SETUP.sh")
+    container.exec("touch /home/svn/svn-project/REMOTE_SETUP.sh")
+    container.exec("svn mkdir file:///home/svn/svn-project/trunk file:///home/svn/svn-project/branches file:///home/svn/svn-project/tags file:///home/svn/svn-project/LOCAL_SETUP.sh file:///home/svn/svn-project/REMOTE_SETUP.sh -m 'first commit'")
+
+    wait_for_tcp_port('localhost', http_port)
+
+
+def get_svn_container(svn_repos_path: Path) -> tuple['DockerContainer', int]:
+    """Get a running SVN container and its HTTP port."""
+    http_port = get_free_port()
+    # noinspection PyProtectedMember
+    container_instance = _create_svn_container(svn_repos_path, http_port)
     return container_instance, http_port
 
 

